utils/tmp_test.py

```diff
@@ -0,0 +1,17 @@
+'''
+step 3
+Add a full traceback to exception to make sure the variable in the scope is printed in an exception.
+'''
+
+def get_message_text(message_struct: dict):
+    message_text = message_struct['text']
+    if type(message_text) in (str, bytes):
+        pass
+    elif type(message_text) == list:
+        message_text = message_text[0]
+    else:
+        message_text = ""
+    if type(message_text) != str:
+        raise AttributeError(f'{message_text} is not a string. Cant use lower method on it')
+
+    return message_text.lower().strip()
```

Comment:
It's a programming religion dogma to leave a trailing space at the end 🤣 
At some point it was the compiler should have known where the code stops. Some people also code in vim or emacs that also love to go to the end of the file and not mess up any code.

But now it's just a thing we all do

Reply:
gotcha